src/systems/world_generation/WorldGeneration.ts

```diff
@@ -0,0 +1,171 @@
+import forest_path from "../../assets/environment/tiles/forest.png";
+import grass_path from "../../assets/environment/tiles/grass.png";
+import mountain_path from "../../assets/environment/tiles/mountains.png";
+import ocean_path from "../../assets/environment/tiles/ocean_deep.png";
+import { randomInt } from "../../helpers/math/MathHelper";
+
+// Define enum for tile types
+export enum TerrainType {
+    Forest,
+    Grass,
+    Mountain,
+    Ocean
+}
+
+// Define a class for the tile
+export class Tile {
+
+    terrainType: TerrainType
+
+    constructor(terrainType: TerrainType) {
+        this.terrainType = terrainType;
+    }
+}
+
+const renderingConfig = {
+    tileSpriteWidth: 32,
+    tileSpriteHeight: 34,
+    horizontallyDisplayedTileCount: 20,
+}
+
+export const getTileTypeKey = (type: TerrainType) => {
+    switch (type) {
+        case TerrainType.Forest:
+            return "Forest";
+        case TerrainType.Grass:
+            return "Grass";
+        case TerrainType.Mountain:
+            return "Mountain";
+        case TerrainType.Ocean:
+            return "Ocean";
+    }
```

Comment:
also if we had a record/array of these, we wouldn't need to get keys - we would already have them or could just loop through them